ckanext/versioning/blueprints.py

```diff
@@ -106,9 +106,9 @@ def changes(id):
 
 versioning.add_url_rule('/dataset/<id>/version/changes', view_func=changes)
 versioning.add_url_rule('/dataset/<package_id>/show', view_func=show)
-versioning.add_url_rule('/dataset/<package_id>/show/<revision_ref>', view_func=show)
+versioning.add_url_rule('/dataset/<package_id>/show/<revision_ref>/<tag_name>', view_func=show)
 versioning.add_url_rule('/dataset/<package_id>/resource/<resource_id>', view_func=resource_show)
-versioning.add_url_rule('/dataset/<package_id>/show/<revision_ref>/<resource_id>', view_func=resource_show)
+versioning.add_url_rule('/dataset/<package_id>/show/<revision_ref>/<resource_id>/<tag_name>', view_func=resource_show)
```

Comment:
This makes specifying `tag_name` in `show` URLs mandatory (because otherwise the route won't be matched). It means we can't re-use the same route or a shorter route to show a revision by ID and not by tag, and it means URL semantics change. 

The whole point of `revision_ref` is that it *can* be both a revision ID *or* a tag name. We do not need to change the URL structure at all, just ensure that the action can take either (maybe it already does), and add some extra information on the tag if what we got is a tag. If you are not sure how to do it, let's find the time to discuss.